extensions/have-a-cookie/src/util/chrome-cookie-helper.js

```diff
@@ -0,0 +1,523 @@
+
+	cookies.forEach(function (cookie, index) {
+
+		out += cookie.name + '=' + cookie.value;
+		if (cookieLength > index + 1) {
+			out += '; ';
+		}
+
+	});
+
+	return out;
+
+}
+
+function convertRawToJar(cookies, uri) {
+
+	var jar = new request.jar();
+
+	cookies.forEach(function (cookie, index) {
+
+		var jarCookie = request.cookie(cookie.name + '=' + cookie.value);
+		if (jarCookie) {
+			jar.setCookie(jarCookie, uri);
+		}
+
+	});
+
+	return jar;
+
+}
+
+function convertRawToSetCookieStrings(cookies) {
+
+	var cookieLength = cookies.length,
+	strings = [];
+
+	cookies.forEach(function(cookie, index) {
+
+		var out = '';
+
+		var dateExpires = new Date(convertChromiumTimestampToUnix(cookie.expires_utc) * 1000);
+
+		out += cookie.name + '=' + cookie.value + '; ';
+		out += 'expires=' + tough.formatDate(dateExpires) + '; ';
+		out += 'domain=' + cookie.host_key + '; ';
+		out += 'path=' + cookie.path;
+
+		if (cookie.is_secure) {
+			out += '; Secure';
+		}
+
+		if (cookie.is_httponly) {
+			out += '; HttpOnly';
+		}
+
+		strings.push(out);
+
+	});
+
+	return strings;
+
+}
+
+function convertRawToPuppeteerState(cookies) {
+
+	let puppeteerCookies = [];
+	
+	cookies.forEach(function(cookie, index) {
+		const newCookieObject = {
+			name: cookie.name,
+			value: cookie.value,
+			expires: cookie.expires_utc,
+			domain: cookie.host_key,
+			path: cookie.path
+		}
+		if (cookie.is_secure) {
+			newCookieObject['Secure'] = true
+		}
+		if (cookie.is_httponly) {
+			newCookieObject['HttpOnly'] = true
+		}
+		puppeteerCookies.push(newCookieObject)
+	})
+
+	return puppeteerCookies;
+}
+
+function convertRawToObject(cookies) {
+
+	var out = {};
+
+	cookies.forEach(function (cookie, index) {
+		out[cookie.name] = cookie.value;
+	});
+
+	return out;
+
+}
+
+function decryptAES256GCM(key, enc, nonce, tag) {
+	const algorithm = 'aes-256-gcm';
+	const decipher = crypto.createDecipheriv(algorithm, key, nonce);
+	decipher.setAuthTag(tag);
+	let str = decipher.update(enc,'base64','utf8');
+	str += decipher.final('utf-8');
+	return str;
+}
+
+/*
+
+Possible formats:
+curl - Netscape HTTP Cookie File contents usable by curl and wget http://curl.haxx.se/docs/http-cookies.html
+jar - request module compatible jar https://github.com/request/request#requestjar
+set-cookie - Array of set-cookie strings
+header - "cookie" header string
+puppeteer - array of cookie objects that can be loaded straight into puppeteer setCookie(...)
+object - key/value of name/value pairs, overlapping names are overwritten
+
+*/
+
+const getCookies = async (uri, format, callback, profile) => {
+
+	profile ? profile : profile = 'Default'
+
+	if (process.platform === 'darwin') {
+
+		path = process.env.HOME + `/Library/Application Support/Google/Chrome/${profile}/Cookies`;
+		ITERATIONS = 1003;
+
+	} else {
+
+		return callback('Only Mac, Windows, and Linux are supported.');
+
+	}
+
+	
+
+	if (format instanceof Function) {
+		callback = format;
+		format = null;
+	}
```

Comment:
This code is also at line 420-437 - could maybe be a new function to keep it dry

Reply:
Actually as Raycast is only available on Mac, there is no need to have this here

Reply:
Correct :)